src/main/java/cz/smarteon/loxone/app/state/ControlState.java

```diff
@@ -0,0 +1,54 @@
+package cz.smarteon.loxone.app.state;
+
+import cz.smarteon.loxone.Loxone;
+import cz.smarteon.loxone.app.Control;
+import cz.smarteon.loxone.message.ControlCommand;
+import cz.smarteon.loxone.message.TextEvent;
+import cz.smarteon.loxone.message.ValueEvent;
+import lombok.RequiredArgsConstructor;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Base class for all the controlStates in loxone application
+ */
+@RequiredArgsConstructor
+public abstract class ControlState<T extends Control> {
+
+    /**
+     * The webSocket connection to communicate with the loxone miniserver.
+     */
+    protected final Loxone loxone;
+
+    /**
+     * The control that this state refers to.
+     */
+    protected final T control;
+
+    /**
+     * Method that accepts ValueEvent from the miniserver to update the internal state.
+     * @param event value event received (should not be null)
+     */
+    void accept(@NotNull ValueEvent event) {
+        // default implementation
```

Comment:
does the default impl. make any sense? Isn't it better to have the method abstract?

Reply:
I assumed the state classes would not always be interested in TextEvents & ValueEvents and to avoid an empty accept method in the implementation i've provided the default implementation (doing nothing with the event) in the abstract class.